siliconcompiler/utils.py

```diff
@@ -23,3 +25,19 @@ def copytree(src, dst, ignore=[], dirs_exist_ok=False, link=False):
             os.link(srcfile, dstfile)
         else:
             shutil.copy2(srcfile, dstfile)
+
+def insecure_md5(data):
+    '''Wrapper around hashlib md5 function to enable use on FIPS compliant
+    platforms.
+
+    This function SHOULD NOT be used for cryptography/security-related purposes.
+    '''
+    encoded_data = data.encode('utf-8')
+
+    if sys.version_info[0] == 3 and sys.version_info[1] >= 9:
+        # usedforsecurity flag added in Python 3.9
+        return hashlib.md5(encoded_data, usedforsecurity=False).hexdigest()
+    else:
+        # Regular md5 call seems to work on FIPS systems when using Python
+        # versions older than 3.9.
```

Comment:
Thanks for taking a look at this; however, we are running python 3.6.8

Reply:
Huh, interesting. I tested this out on a FIPS-compliant Ubuntu 18.04 VM with Pythons 3.6-3.10 and it seemed to work. Passing in this flag on the older Python versions resulted in an exception. 

Could you send me some more info about your environment? Which OS you're running and where your Python installation came from if you know (distro package manager, built from source, etc.) would be helpful. If you'd prefer to follow up via email that's fine too (noah@zeroasic.com). 

Thanks!

Reply:
Actually, never mind - we'll go with using sha1() instead. No reason for us not to use it here, and it will avoid any pitfalls related to platform/Python version compatibility.